src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2149,32 +2208,26 @@ int wm_vuldet_check_agent_vulnerabilities(agent_software *agents, wm_vuldet_flag
         // Reset the tables before scanning each agent
         wm_vuldet_reset_tables(db);
 
-        int request = 0;
-        // First step: collect its software
-        if (result = wm_vuldet_get_software_info(agents_it, db, min_full_scan_interval, &request), result == OS_INVALID) {
-            mterror(WM_VULNDETECTOR_LOGTAG, VU_GET_SOFTWARE_ERROR, atoi(agents_it->agent_id));
+        // Collect agent software
+        if (OS_SUCCESS != wm_vuldet_collect_agent_software(agents_it, db, &scan_ctx)) {
+            mterror(WM_VULNDETECTOR_LOGTAG, VU_GET_SOFTWARE_ERROR, scan_ctx.agent_id);
             break;
         }
 
-        // result == 2 skips the agent
-        // is used when no hotfixes are available or no packages have been marked for scanning
-        if (result != 2) {
-            if (VU_SOFTWARE_FULL_REQ == request) {
-                //Clean the vulnerabilities in the agent database
-                int sock = wm_vuldet_get_wdb_socket();
-                if (OS_INVALID == wdb_agents_vuln_cves_clear(atoi(agents_it->agent_id), &sock)) {
-                    mtwarn(WM_VULNDETECTOR_LOGTAG, "Failed to clear vulnerabilities from the agent %s database", agents_it->agent_id);
-                }
-            }
-            // Second step: find and report vulnerabilities
-            if (wm_vuldet_report_agent_vulnerabilities(db, agents_it, flags) < 0) {
-                mterror(WM_VULNDETECTOR_LOGTAG, VU_REPORT_ERROR, atoi(agents_it->agent_id), sqlite3_errmsg(db));
-                break;
-            }
+        // Find and report agent vulnerabilities
+        if (OS_SUCCESS != wm_vuldet_find_agent_vulnerabilities(db, agents_it, &vuldet->flags, &scan_ctx)) {
+            mterror(WM_VULNDETECTOR_LOGTAG, VU_REPORT_ERROR, scan_ctx.agent_id, sqlite3_errmsg(db));
+            break;
+        }
+
+        // Find and report obsolete vulnerabilities
+        if (OS_SUCCESS != wm_vuldet_find_obsolete_vulnerabilities(&scan_ctx) ) {
+            mterror(WM_VULNDETECTOR_LOGTAG, "The agent '%.3d' obsolete vulnerability, could not be processed", scan_ctx.agent_id);
```

Comment:
Maybe, we could change the structure of the sentence.

Reply:
Do you mean to change the log message to something like: _The agent '%.3d' obsolete vulnerability could not be processed_ ?

Reply:
Yes, or something similar, it's OK